nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+        default = 2;
+        description = "Number of FastCGI processes to fork.";
+      };
+    };
+
+    mta = {
+      type = mkOption {
+        type = enum [ "postfix" "none" ];
+        default = "postfix";
+        description = ''
+          Mail transfer agent (MTA) integration. Use `none` if you want to configure it yourself.
+
+          The `postfix` integration sets up local Postfix instance that will pass incoming
+          messages from configured domains to Sympa. You still need to configure at least
+          outgoing message handling using e.g. <option>services.postfix.relayHost</option>.
+        '';
+      };
+    };
+
+    settings = mkOption {
+      type = attrsOf (either (either str int) bool);
+      default = {};
+      example = {
+        default_home = "lists";
+        viewlogs_page_size = 50;
+      };
+      description = ''
+        The sympa.conf configuration file as key value set.
+      '';
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    services.sympa.settings = (mapAttrs (_: v: mkDefault v) {
+      domain     = if cfg.mainDomain != null then cfg.mainDomain else (head fqdns);
+      listmaster = concatStringsSep "," cfg.listMasters;
+      lang       = cfg.lang;
+
+      home       = "${dataDir}/list_data";
+      arc_path   = "${dataDir}/arc";
+      bounce_path = "${dataDir}/bounce";
+
+      sendmail = "/run/wrappers/bin/sendmail";
+
+      db_type = cfg.database.type;
+      db_name = cfg.database.name;
+      db_passwd = "#dbpass#";
+    }
+    // (optionalAttrs (cfg.database.host != null) {
+      db_host = cfg.database.host;
+    })
+    // (optionalAttrs (cfg.database.port != null) {
+      db_port = cfg.database.port;
+    })
+    // (optionalAttrs (cfg.database.user != null) {
+      db_user = cfg.database.user;
+    })
+    // (optionalAttrs (cfg.mta.type == "postfix") {
+      sendmail_aliases = "${dataDir}/sympa_transport";
+      aliases_program  = "${pkgs.postfix}/bin/postmap";
+      aliases_db_type  = "hash";
+    })
+    // (optionalAttrs cfg.web.enable {
+      static_content_path = "${dataDir}/static_content";
+      css_path            = "${dataDir}/static_content/css";
+      pictures_path       = "${dataDir}/static_content/pictures";
+      mhonarc             = "${pkgs.perlPackages.MHonArc}/bin/mhonarc";
+    }));
+
+    environment = {
+      systemPackages = [ pkg ];
+    };
+
+    users.users = optional (cfg.user == "sympa")
+      { name = cfg.user;
+        description = "Sympa mailing list manager user";
+        group =  cfg.group;
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+      };
+
+    users.groups =
+      optional (cfg.group == "sympa")
+      { name = cfg.group;
+      };
+
+    warnings = optional (cfg.database.password != "")
+      ''config.services.sympa.database.password will be stored as plaintext
+        in the Nix store. Use database.passwordFile instead.'';
```

Comment:
If you have a warning suggesting this, it's probably better to just not provide the `password` option at all.

Reply:
Works for me. I copied this from `gitea.nix` and it's also in two more files btw.

Reply:
And it's time to remove all instances like this. I'll add that to my list of things to accomplish before 19.09. If people really want to shoot themselves in the foot (or just test something) there is always `password = pkgs.writeText "password.txt" "test-password";`